rolepermissions/roles.py

```diff
@@ -172,6 +176,21 @@ def get_default(cls, permission_name):
         return cls.available_permissions[permission_name]
 
 
+def get_or_create_permission(codename, name=camel_or_snake_to_title):
+    """
+    Get a Permission object from a permission name.
+    @:param codename: permission code name
+    @:param name: human-readable permissions name (str) or callable that takes codename as argument and returns str
+    """
+    user_ct = ContentType.objects.get_for_model(get_user_model())
+    # Careful here - don't use name to lookup existing permissions
+    try:
```

Comment:
Should we use a [`get_or_create`](https://docs.djangoproject.com/en/1.11/ref/models/querysets/#get-or-create) with `codename` in the `defaults` here?

Reply:
Like...

    perm, created = Permission.objects.get_or_create(content_type=user_ct, codename=codename)
    if created:  
        perm.name=name(codename) if callable(name) else name
        perm.save()
?

Reply:
@powderflask `get_or_create` allows a `defaults` value that will to the trick. See here https://docs.djangoproject.com/en/1.11/ref/models/querysets/#get-or-create
```
name = name(codename) if callable(name) else name
perm, created = Permission.objects.get_or_create(
    content_type=user_ct, codename=codename
    defaults={'name': name})
```

Reply:
Ha ha - new to me, go figure !! :-P  so simple, duh...

    return Permission.objects.get_or_create(content_type=user_ct, codename=codename, defaults={'name':name(codename) if callable(name) else name})

Thanks.